GoogleDataTransportCCTSupport/GoogleDataTransportCCTSupport/Classes/GDTCCTUploader.m

```diff
@@ -34,8 +48,94 @@ + (instancetype)sharedInstance {
   return sharedInstance;
 }
 
-- (void)uploadPackage:(nonnull GDTUploadPackage *)package
-           onComplete:(nonnull GDTUploaderCompletionBlock)onComplete {
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    _uploaderQueue = dispatch_queue_create("com.google.GDTCCTUploader", DISPATCH_QUEUE_SERIAL);
+    NSURLSessionConfiguration *config = [NSURLSessionConfiguration defaultSessionConfiguration];
+    _uploaderSession = [NSURLSession sessionWithConfiguration:config];
+  }
+  return self;
+}
+
+- (NSURL *)defaultServerURL {
+  static NSURL *defaultServerURL;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    // These strings should be interleaved to construct the real URL. This is just to (hopefully)
+    // fool github URL scanning bots.
+    const char *p1 = "hts/frbslgiggolai.o/0clgbth";
```

Comment:
Ugh! Security by obscurity.  Maybe necessary but seems ugly.

Reply:
This isn't really security, because we expect all humans to be able to see what's happening here--all we're really attempting to do is prevent bots that scan github repos for URLs from automating network requests directed at that URL.